NO-TICKET Guide/Core doc cleanup (#1722)
CMSgov/design-system: CONTRIBUTING.md

Here are a few examples of how the design system favors clarity over succinctness with its CSS class names:

| Design system (Good)      | Abstract names (Bad) | Description                                                     |
| ------------------------- | -------------------- | --------------------------------------------------------------- |
| 👍 `ds-u-margin-right--1` | 🙅 `mr1`             | A utility class, applies a right margin that is 1x the baseline |
| 👍 `ds-c-button`          | 🙅 `btn`             | A button component                                              |

### Favor generalized names

Avoid naming that indicates a specific use-case or styling, e.g. `snippet` rather than `search-snippet`.

The names and phrasing of the design system's parts should also still make sense when a different theme is applied, e.g. `primary` rather than `blue`.

### Variations exist only when they can't be achieved with a utility class

> The design system includes utility classes for modifying the most common visual traits, like spacing, size, and color. You may want to offer variations of a component that involves changing one of these traits. When you do, you should first see if you can accomplish the variation by using one of the utility classes. If you can, then don't create a new CSS ruleset to accomplish something another ruleset already does.

Reviewer: ```suggestion
The design system includes utility classes for modifying the most common visual traits, like spacing, size, and color. You may want to offer variations of a component that involves changing one of these traits. When you do, you should first see if users of the design can accomplish the variation by applying one of the utility classes to your component.
```

Reviewer: I don't want to encourage people to use utility classes _inside_ of components. That's something we just finished cleaning up!
